src/countBy.ts

```diff
@@ -0,0 +1,57 @@
+import { purry } from "./purry";
+
+/**
+ * Categorize and count elements in an array using a defined callback function. The transformation function applies to each element in the array to determine its category and then counts how many elements fall into each category.
+ *
+ * @param data - The array.
+ * @param fn - The transformation function.
+ * @signature
+ *   R.countBy(array, fn)
```

Comment:
The param is called data, so let's keep the naming consistent

```suggestion
 *   R.countBy(data, fn)
```